### PR TITLE
double proposal detector

### PR DESCRIPTION
**Pull request by shayzluf**
[Part of] #5116

### CODE REVIEW

File: slasher/detection/service.go [hunk 1]
```diff
@@ -10,6 +10,8 @@ import (
 	"github.com/prysmaticlabs/prysm/slasher/db"
 	"github.com/prysmaticlabs/prysm/slasher/detection/attestations"
 	"github.com/prysmaticlabs/prysm/slasher/detection/attestations/iface"
+	"github.com/prysmaticlabs/prysm/slasher/detection/proposals"
+	iface2 "github.com/prysmaticlabs/prysm/slasher/detection/proposals/iface"
```
Comment (rauljordan) [inline slasher/detection/service.go]: proposerIface instead

File: slasher/detection/proposals/detector.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/helpers"
+	"github.com/prysmaticlabs/prysm/slasher/db"
+	status "github.com/prysmaticlabs/prysm/slasher/db/types"
+	"github.com/prysmaticlabs/prysm/slasher/detection/proposals/iface"
+	"go.opencensus.io/trace"
+)
+
+var _ = iface.ProposalsDetector(&ProposeDetector{})
```
Comment (rauljordan) [inline slasher/detection/proposals/detector.go]: Can this assertion live in the test file?

File: slasher/detection/proposals/detector_test.go [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+	"crypto/rand"
+	"reflect"
+	"testing"
+
+	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
+	"github.com/prysmaticlabs/prysm/shared/params"
+	testDB "github.com/prysmaticlabs/prysm/slasher/db/testing"
+)
+
+func signedBlockHeader(slot uint64, proposerIdx uint64) (*ethpb.SignedBeaconBlockHeader, error) {
```
Comment (rauljordan) [inline slasher/detection/proposals/detector_test.go]: Lets move these unexported helpers to the bottom of the file

File: slasher/detection/proposals/detector.go [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+// DetectDoublePropose detects double proposals given a block by looking in the db.
+func (dd *ProposeDetector) DetectDoublePropose(
+	ctx context.Context,
+	incomingBlk *ethpb.SignedBeaconBlockHeader,
+) (*ethpb.ProposerSlashing, error) {
+	ctx, span := trace.StartSpan(ctx, "detector.DetectDoublePropose")
+	defer span.End()
+	epoch := helpers.SlotToEpoch(incomingBlk.Header.Slot)
+	//validatorIdx:=blk.Header.ProposerIndex
```
Comment (rauljordan) [inline slasher/detection/proposals/detector.go]: Maybe prefix this with a TODO for now pointing to a tracking issue

File: slasher/detection/service.go (hunk 1 above)
Comment (shayzluf) [reply slasher/detection/service.go]: thanks

File: slasher/detection/proposals/detector.go (hunk 2 above)
Comment (shayzluf) [reply slasher/detection/proposals/detector.go]: done

File: slasher/detection/proposals/detector.go [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+// DetectDoublePropose detects double proposals given a block by looking in the db.
+func (dd *ProposeDetector) DetectDoublePropose(
+	ctx context.Context,
+	incomingBlk *ethpb.SignedBeaconBlockHeader,
+) (*ethpb.ProposerSlashing, error) {
+	ctx, span := trace.StartSpan(ctx, "detector.DetectDoublePropose")
+	defer span.End()
+	epoch := helpers.SlotToEpoch(incomingBlk.Header.Slot)
+	//TODO(#5119) remove constand and use input from block header.
+	//validatorIdx:=blk.Header.ProposerIndex
```
Comment (nisdas) [inline slasher/detection/proposals/detector.go]: ? , why is it commented

Comment (shayzluf) [reply slasher/detection/proposals/detector.go]: we still don't have proposer index in the block header

Comment (shayzluf) [reply slasher/detection/proposals/detector.go]: would like to use stub till we get v0.11 merged

**Pull request merged**